content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,272 @@
+
+  2. **Set the highest fee** you are comfortable with. Crucially, you should remember that *this doesn't mean you'll **pay** that on every transaction*. You will only pay whatever is necessary to get you into the ledger: under normal (non-surge) circumstances, even with a higher *maximum* fee set, you will pay the *standard* fee (100 stroops as of this writing).
+
+The latter strategy balances simplicity, efficacy, and convenience, but unless you set your maximum high enough simply *never* be exceeded, can still lead to failures. The former strategy can provide more-reliable submissions by allowing tighter guarantees about whether or not a transaction will be accepted. In general, though, it's important to track fee costs: if the network saturates beyond your maximum willingness to pay, perhaps waiting for activity to die down or periodically retrying with the same fee is the best approach for your use case.
+
+If you want to match a fee error **exactly**, you might write something like this:
+
+<CodeExample>
+
+```js
+function isFeeError(error) {
+  return
+    error.response !== undefined && 
+    error.status == 400 &&
+    error.extras &&
+    error.extras.result_codes.transaction == sdk.TX_INSUFFICIENT_FEE;
+}
+```
+
+</CodeExample>
+
+Of course, there are much more streamlined ways to *combine* errors together, but this will be used below to demonstrate the (very) specific check.
+
+### Example: Paying 10% above average
+Suppose we want a fairly conservative fee-paying strategy: we're only willing to pay a 10% higher fee than the average transaction paid.
+
+<CodeExample>
+
+```js
+// when submitting any transaction, first query fee stats
+server.feeStats().then(function (response) {
+  let tx = base.TransactionBuilder(someAccount, {
+    fee: parseFloat(response.fee_charged.p50) * 1.10).toFixed(2),
+    // ...
+  });
+  // ...build the rest of the tx... 
+  tx.sign(someAccount);
+  return server.submitTransaction(tx);
+});
+```
+
+</CodeExample>
+
+### Bumping fees on past transactions
+It's possible that even with the liberal fee-paying policy, your transaction fails to make it into the ledger due to insufficient funds or untimely surges. Resolving this problem is exactly the goal of a [fee-bump transaction](https://developers.stellar.org/docs/glossary/fee-bumps/); the following snippet shows how you can resubmit a transaction with a higher fee given that you have the original [transaction envelope](https://developers.stellar.org/docs/glossary/transactions/#transaction-envelopes):
+
+<CodeExample>
+
+```js
+// Let `lastTx` be some transaction that fails submission due to high fees, and
+// `lastFee` be the maximum fee (expressed as a float) willing to be paid by 
+// `account` for `lastTx`.
+server.submitTransaction(lastTx).catch(function (error) {
+  if (isFeeError(error)) {
+    let bump = sdk.TransactionBuilder.buildFeeBumpTransaction(
+      account, (lastFee * 10).toFixed(2), lastTx, server.networkPassphrase
+    );
```

Comment:
This isn't clear to me. The fee should be an integer (in stroops) but we use `toFixed` here. Also adding a suggestion that will be more readable:
```suggestion
    let bump = sdk.TransactionBuilder.buildFeeBumpTransaction(
      account, // Account that will pay the fee
      (lastFee * 10).toFixed(2), // New fee
       lastTx,
       server.networkPassphrase
    );
```

Reply:
Good point! It's supposed to be an integer but encoded as a string; I'll rectify this! `.toFixed` is necessary to avoid long decimals (e.g. `1000.000000001`) but you're right that there shouldn't be a decimal in the first place.